runtime/src/mem/jemalloc/mem-jemalloc.c

```diff
@@ -47,11 +47,12 @@
 // the performance difference in a playground just to see if it's worth
 // investigating further.
 
-// TODO put these into a struct?
-static void*  heap_base = NULL;
-static size_t heap_size = 0;
-static size_t cur_heap_offset = 0;
-static pthread_mutex_t chunk_alloc_lock;
+static struct shared_heap {
+  void* base;
+  size_t size;
+  size_t cur_offset;
+  pthread_mutex_t alloc_lock;
+} heap; // static, will be "zero" initialized automatically
```

Comment:
Remember that we only use the lock when we have the shared heap. I originally did the static init in https://github.com/chapel-lang/chapel/pull/3264, but changed it because of your slight preference for the dynamic init.

If we use the static init, we're still supposed to call mutex destroy so we'd probably just end up with the static init, and always destroying it in `chpl_mem_layerExit()`